### PR TITLE
Change behaviour of theme_update to be more like an update

### PR DESCRIPTION
**Pull request by oneillkza**
Per discussion in #1489, this changes `theme_update()` to use the `+` operator, so that unspecified attributes of the element being set are taken from the current theme, rather than `NULL`ing them out. I've added `theme_replace()` with the same behaviour as the current `theme_update()`, in case that behaviour is desired (though I'm hard-pressed to think of a use case).

I've also added some verbiage to the documentation clarifying what these do, along with examples.

### CODE REVIEW

**Comment by hadley** [comment]
@wch do you remember the motivation for using `%+replace%` here?

File: R/theme.r [hunk 1]
```diff
@@ -15,19 +21,40 @@
...
 theme_update <- function(...) {
   # Make a call to theme, then add to theme
+  theme_set(theme_get() + theme(...))
+}
+
+#' @rdname theme_update
+#' @export
+
+theme_replace <- function(...) {
+  # Make a call to theme, then add to theme
```
Comment (hadley) [inline R/theme.r]: Can you please remove this comment? (and in the previous fun)

File: R/theme.r [hunk 2]
```diff
@@ -15,19 +21,40 @@
...
 #'  theme(legend.background = element_rect(fill = "white", colour = "white", size = 3))
+#'
 theme_update <- function(...) {
   # Make a call to theme, then add to theme
+  theme_set(theme_get() + theme(...))
+}
+
+#' @rdname theme_update
+#' @export
+
```
Comment (hadley) [inline R/theme.r]: Please remove this empty line

**Comment by hadley** [comment]
Looks good. Just a few minor tweaks and can you please add a bullet point to NEWS?

**Comment by oneillkza** [comment]
Done!

File: NEWS.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # ggplot2 2.0.0.9000
 
+* `theme_update()` now uses the `+` operator instead of `%+replace%`, so that
+  unspecified values will no longer be `NULL`ed out. `theme_replace()`
+  preserves the old behaviour if desired (#1519).
```
Comment (hadley) [inline NEWS.md]: Could you please cite your github user name in the brackets?

**Comment by oneillkza** [comment]
Done!

**Comment by hadley** [comment]
Thanks!

**Pull request merged**